tests/queries/0_stateless/03202_system_load_primary_key.sql

```diff
@@ -0,0 +1,75 @@
+-- Tags: no-parallel
+
+-- Tests statement SYSTEM LOAD PRIMARY KEY
+
+DROP TABLE IF EXISTS tab1;
+DROP TABLE IF EXISTS tab2;
+
+-- Create the test tables
+CREATE TABLE tab1 (id Int32, val String) ENGINE = MergeTree() ORDER BY id;
+CREATE TABLE tab2 (id Int32, val String) ENGINE = MergeTree() ORDER BY id;
+
+-- Insert data into both tables
+INSERT INTO tab1 VALUES (1, 'a'), (2, 'b'), (3, 'c');
+INSERT INTO tab2 VALUES (1, 'x'), (2, 'y'), (3, 'z');
+
+-- Check primary key memory before loading (this checks if it's not loaded yet) for both tables
+SELECT
+    table,
+    round(primary_key_bytes_in_memory, -7),
+    round(primary_key_bytes_in_memory_allocated, -7)
+FROM system.parts
+WHERE
+    database = currentDatabase()
+    AND table IN ('tab1', 'tab2')
+ORDER BY table;
+
+-- Load primary keys for all tables in the database
+SYSTEM LOAD PRIMARY KEY;
+
+-- Verify primary key memory after loading for both tables
+-- Ensure .reference file has non-zero values here to reflect expected primary key loading
```

Comment:
If I understand correctly,
- the check in l. 17 is reflected in l. 1-2 of the .reference file and,
- the check in l. 32 (here) is reflected in l. 3-4 of the .reference file.

L. 31 says the .reference file has a non-zero value but the values on l. 3-4 are zero. I am confused.

Reply:
It is hard to associate the results in the .reference file with the SELECTs in the .sql file.

Can you please rewrite your checks in the following manner?

```sql
-- Check primary key memory before loading (this checks if it's not loaded yet) for both tables
SELECT
    table,
    round(primary_key_bytes_in_memory, -7),
    round(primary_key_bytes_in_memory_allocated, -7)
FROM system.parts
WHERE
    database = currentDatabase()
    AND table IN ('tab1', 'tab2')
ORDER BY table;

-- in the .reference file:
tab1    0    0
tab2    0    0
```

becomes

```sql
SELECT `Expect that the primary key is not loaded into memory before it is first accessed';
SELECT
    table,
    round(primary_key_bytes_in_memory, -7),
    round(primary_key_bytes_in_memory_allocated, -7)
FROM system.parts
WHERE
    database = currentDatabase()
    AND table IN ('tab1', 'tab2')
ORDER BY table;

-- in the reference file:
Expect that the primary key is not loaded into memory before it is first accessed
tab1    0    0
tab2    0    0
```

Reply:
@Zawa-ll the reason why there are 0s after loading the primary keys is because you're using `round(primary_key_bytes_in_memory, -7)` as in my `03128_system_unload_primary_key`. However, both in my test and in the original test I took it from `02993_lazy_index_loading` the amount of data inserted is huge.

In other words: either you adapt/remove the `round(..., -7)` or you need to insert more data.

Reply:
Thank y'all! I will check and fix this issue!!